scapy/contrib/automotive/xcp/scanner.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Tabea Spahn <tabea.spahn@e-mundo.de>
+# This program is published under a GPLv2 license
+
+# scapy.contrib.description = XCPScanner
+# scapy.contrib.status = loads
+
+from typing import Optional, List, Tuple
+
+from scapy.contrib.automotive.xcp.cto_commands_master import \
+    TransportLayerCmd, TransportLayerCmdGetSlaveId, Connect
+from scapy.contrib.automotive.xcp.xcp import CTORequest, XCPOnCAN
+from scapy.contrib.cansocket_native import CANSocket
+
+
+class XCPScannerResult:
+    def __init__(self, slave_id, response_id):
+        self.slave_id = slave_id
+        self.response_id = response_id
+
+
+class XCPOnCANScanner:
+    """
+    Scans for XCP Slave on CAN
+    """
+
+    def __init__(self, can_socket, id_range=None,
+                 sniff_time=0.1, verbose=False):
+        # type: (CANSocket, Optional[Tuple[int, int]], Optional[float], Optional[bool]) -> None # noqa: E501
+
+        """
+        Constructor
+        :param can_socket: Can Socket with XCPonCAN as basecls for scan
+        :param id_range: CAN id range to scan
+        :param sniff_time: time the scan waits for a response
+                           after sending a request
+        """
+        self.__socket = can_socket
+        self.id_range = id_range or (0, 0x7ff)
```

Comment:
You could use `range(0, 0x800)` here and later just ask for id_range. this has the advantage, that you could treat id_range as a `Iterable[int]` which gives the freedom to use lists, generators or ranges as argument type